recipes/open62541/all/conanfile.py

```diff
@@ -0,0 +1,307 @@
+                raise ConanInvalidConfiguration(
+                    "Lower Open62541 versions than 1.1.0 do not fully support multithreading")
+
+            if self.options.web_socket:
+                raise ConanInvalidConfiguration(
+                    "Lower Open62541 versions than 1.1.0 do not fully support websockets")
+
+            if self.options.cpp_compatible:
+                raise ConanInvalidConfiguration(
+                    "Lower Open62541 versions than 1.1.0 are not cpp compatible due to -fpermisive flags")
+
+            if self.settings.compiler == "clang" and tools.Version(self.settings.compiler.version) >= "9":
+                raise ConanInvalidConfiguration(
+                    "Open62541 does not support Clang 9.x.x and later compiler versions")
+
+        if self.settings.compiler == "clang":
+            if tools.Version(self.settings.compiler.version) < "5":
+                raise ConanInvalidConfiguration(
+                    "Older clang compiler version than 5.0 are not supported")
+
+        if self.options.pub_sub != False and self.settings.os != "Linux":
+            raise ConanInvalidConfiguration(
+                "PubSub over Ethernet is not supported for your OS!")
+
+        if self.options.web_socket:
+            self.options["libwebsockets"].with_ssl = self.options.encryption
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+        if not self.options.cpp_compatible:
+            del self.settings.compiler.cppstd
+            del self.settings.compiler.libcxx
+
+    def source(self):
+        archive_name = self.name + "-" + self.version
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(archive_name, self._source_subfolder)
+
+        submodule_filename = os.path.join(
+            self.recipe_folder, 'submoduledata.yml')
+        with open(submodule_filename, 'r') as submodule_stream:
+            submodules_data = yaml.load(submodule_stream)
+            for path, submodule in submodules_data["submodules"][self.version].items():
+                filename = os.path.basename(submodule["url"])
+                archive_name = submodule["archive_pattern"].format(
+                    version=os.path.splitext(filename.replace('v', ''))[0])
+
+                submodule_data = {
+                    "url": submodule["url"],
+                    "sha256": submodule["sha256"]
+                }
+
+                tools.get(**submodule_data)
+                submodule_source = os.path.join(self._source_subfolder, path)
+                tools.rmdir(submodule_source)
+                os.rename(archive_name, submodule_source)
+
+    def _get_log_level(self):
+        return {
+            "Fatal": "600",
+            "Error": "500",
+            "Warning": "400",
+            "Info": "300",
+            "Debug": "200",
+            "Trace": "100",
+            "PackageOption": "300"
```

Comment:
@saukijan
Hi, was this `PackageOption` value intended?
I'm asking because I add the logging_level option for pro-mdnsd

Reply:
No, I do not remeber adding this option, it seems to be a placeholder for the default logging value, but I don't see that it is used in the recipie. It might habe come from some other contributor before my code refactor

You can remove it and see if the recipie works without it. It should work just fine ,if I remeber the open62541 cmake options

Reply:
Thank you. I've done that: https://github.com/conan-io/conan-center-index/pull/27258